server/controllers/roleController.js

```diff
@@ -4,37 +4,33 @@ const {errorMessage} = require("../utils/error");
 
 let RoleController = function (roleRepository) {
 // Create a new role
-  this.create = function (req, res) {
+  this.create = async function (req, res) {
     const errors = validationResult(req);
     if (!errors.isEmpty()) {
       return res.status(400).json({message: "Invalid request", errors: errors.array()});
     }
     // Check if admin
     if (!req.user.isAdmin) {
-      res.status(401).json({message: "Only admins can add roles"});
-      return;
+      return res.status(401).json({message: "Only admins can add roles"});
     }
     // Check if valid hex string
     if (req.body.colour) {
       if (!/^#[0-9A-F]{6}$/i.test(req.body.colour)) {
```

Comment:
nit: Would be nice to document this regex.

Reply:
Extracted to a named method, in next commit